plugin/src/main/scala/com/lightbend/sbt/JavaFormatterPlugin.scala

```diff
@@ -41,7 +42,9 @@ object JavaFormatterPlugin extends AutoPlugin {
       "Execute the javafmtCheck task for all configurations in which it is enabled. " +
       "(By default this means the Compile and Test configurations.)")
     val javafmtOnCompile = settingKey[Boolean]("Format Java source files on compile, on by default.")
-    val javafmtAospStyle = settingKey[Boolean]("Use AOSP style if set, otherwise Google Java Style")
+    val javafmtOptions = settingKey[JavaFormatterOptions.Style](
+      "Define options like style " +
+      "(Google Java Style or AOSP) or JavaDoc")
```

Comment:
The description doesn't match the type - we can only specify the style here. This should be a `settingKey[JavaFormatterOptions]`.

Reply:
Hi @nigredo-tori, thank you for the review. Last commit was rather a "save" commit than "review" commit :)

I changed the type of the setting from `JavaFormatterOptions.Style` to `JavaFormatterOptions`.
Here's the `build.sbt` configuration to use ASOP style
> ThisBuild / javafmtOptions := JavaFormatterOptions.builder().style(JavaFormatterOptions.Style.AOSP).build()

Isn't it too verbose?

Reply:
IMO, `JavaFormatterOptions` is a good type to have a `settingKey` for, since it contains all the available configuration knobs. If using it directly is cubersome, we can also do something like this:
```scala
val javafmtStyle = settingKey[JavaFormatterOptions.Style](???)
val javafmtOptions = settingKey[JavaFormatterOptions](???)
// Plugin's settings
javafmtOptions := JavaFormatterOptions.builder().style(javafmtStyle.value).build()
```
(this is just an example - I've probably messed up the scopes).

Reply:
> JavaFormatterOptions is a good type to have a settingKey

I agree

> If using it directly is cubersome

Version 1.7 of google-java-format can only configure the style. Version 1.8 can additionally enable Javadoc formatting. I don't think it is cumbersome if an example is provided in the README (`ThisBuild / javafmtOptions := JavaFormatterOptions.builder().style(JavaFormatterOptions.Style.AOSP).build()`)

Reply:
@nigredo-tori I gave a quick look at 
```sbt
val javafmtStyle = settingKey[JavaFormatterOptions.Style](???)
val javafmtOptions = settingKey[JavaFormatterOptions](???)
javafmtOptions := JavaFormatterOptions.builder().style(javafmtStyle.value).build()
```
and I have not found a solution (yet?) to set default values for dependent settings.
What do you think? Should we provide finer settings or keep the current solution?

Reply:
> and I have not found a solution (yet?) to set default values for dependent settings.

Speaking in the context of [this plugin], the defaults should be in `globalSettings`:
```scala
javafmtStyle := ???
```
, and we can build dependent settings from those in `toBeScopedSettings`:
```scala
javafmtOptions := JavaFormatterOptions.builder().style(javafmtStyle.value).build()
```
This way the calling code can set `ThisBuild / javafmtStyle`  and be done with it. SBT does the same thing with `forkOptions`, so I think this should probably be good enough for us.

Reply:
Sure, I was actually looking to also override `javafmtOptions := JavaFormatterOptions.builder().style(javafmtStyle.value).build()` if the calling code set its own options like `ThisBuild / javafmtOptions := JavaFormatterOptions.builder().style(<whatever>).build()`

Reply:
You can do that - there's `.?` and `.??(defaultValue)` syntax for `SettingKey`s and the like, which can be used to set a value iff it was not defined in a more general scope. But I don't think the added complexity is worth it.

Anyway, if we provide a comprehensive set of fine-grained options like `javafmtStyle`, overriding `javafmtOptions` should become a rare case, and should still be reasonably straightforward to handle.